File: CalculatorApi/Models/CalculationResponse.cs
namespace CalculatorApi.Models;

/// <summary>
/// Result returned by the calculator API after a successful computation.
/// </summary>
/// <param name="Number1">The first operand used in the calculation.</param>
/// <param name="Number2">The second operand used in the calculation.</param>
/// <param name="Operation">The operation that was performed.</param>
/// <param name="Symbol">Human-readable operator symbol (e.g. "+", "−", "×", "÷").</param>
/// <param name="Result">The computed result.</param>
/// <param name="Expression">A formatted expression string, e.g. "3 + 4 = 7".</param>
public record CalculationResponse(
    double Number1,
    double Number2,
    OperationType Operation,
    string Symbol,
    double Result,
    string Expression
);

File: java-calculator/src/main/java/com/example/calculator/model/CalculationResponse.java
package com.example.calculator.model;

/**
 * Outgoing calculation response.
 *
 * <p>Uses a Java 16+ {@code record} for an immutable, concise data carrier.
 *
 * @param firstNumber  the first operand echoed back
 * @param secondNumber the second operand echoed back
 * @param operation    the operation performed
 * @param result       the computed result
 * @param expression   a human-readable expression string, e.g. "3.0 + 4.0 = 7.0"
 */
public record CalculationResponse(
        double firstNumber,
        double secondNumber,
        String operation,
        double result,
        String expression) {}
